mappings/net/minecraft/entity/Entity.mapping

```diff
@@ -13,36 +16,48 @@ CLASS ail net/minecraft/entity/Entity
 	FIELD O age I
 	FIELD P insideWater Z
 	FIELD Q waterHeight D
+	FIELD R inWater Z
 	FIELD S inLava Z
+	FIELD T regenTimer I
```

Comment:
This isn't a timer, it's just an int, so it should be `regenTime`, or `timeUntilRegen`. What does "regen" mean though? How foes a non-living entity regen?

Reply:
This is definitely a timer of some kind. It's used in `LivingEntity.baseTick()` (which I can't see).
It is being set to a value of `60` in `SkeletonHorseTrapTriggerGoal` both by `getSkeleton` and by `getHorse`. 
It's set to `10` or `20` in `ClientPlayerEntity.setHealth()`
It is decremented every tick in `ServerPlayerEntity.tick()` (if greater than 0)

        if (this.regenTimer > 0) {
            --this.regenTimer;
        }

It is also referenced in `InGameHub.renderStatusBar` and is thus very clearly closeley related to `health` and `time`.

        int integer3 = MathHelper.ceil(playerEntity2.getHealth());
        boolean boolean4 = this.C > this.ticks && (this.C - this.ticks) / 3L % 2L == 1L;
        long long5 = SystemUtil.getMeasuringTimeMs();
        if (integer3 < this.z && playerEntity2.regenTimer > 0) {
            this.B = long5;
            this.C = this.ticks + 20;
        }
        else if (integer3 > this.z && playerEntity2.regenTimer > 0) {
            this.B = long5;
            this.C = this.ticks + 10;

After some investigation of this method. `B` and `C` are both specific to those lines, however `C` is used to calculate `boolean4` whilst goes on to be used later when rendering health:

            int integer25 = 0;
            if (boolean4) {
                integer25 = 1;
            }

Changing `integer25` to only affects the first `blit` call after that, shifting the texture y position down 9 pixels, which you can probably attribute to the white "flash" hearts create when the player's health is regenerating.

Reply:
The thing is, calling the int a timer is semantically wrong. A timer is an object that keeps track of a time variable. What you call `regenTimer` is not an actual timer, it is the time that is ticked down by the entity. Hence `timeUntilRegen`.

Reply:
Renamed then. :/